lib/spack/llnl/util/filesystem.py

```diff
@@ -293,6 +349,31 @@ def hide_files(*file_list):
             shutil.move(bak, f)
 
 
+def hash_directory(directory):
+    """Hashes recursively the content of a directory.
+
+    Args:
+        directory (path): path to a directory to be hashed
+
+    Returns:
+        hash of the directory content
+    """
+    assert os.path.isdir(directory), '"directory" must be a directory!'
+
+    md5_hash = hashlib.md5()
+
+    # Adapted from https://stackoverflow.com/a/3431835/771663
+    for root, dirs, files in os.walk(directory):
+        for name in sorted(files):
+            filename = os.path.join(root, name)
+            # TODO: if caching big files becomes an issue, convert this to
```

Comment:
I would add that this method is only used for testing purposes. If it were used to hash actual install directories I would definitely request this be changed.

Also: is there some reason you prefer this to dircmp (https://docs.python.org/2/library/filecmp.html)? This is so small I don't really mind it being implemented here as is so I'm just curious.

Reply:
It seems to me that `filecmp` is a better fit to compare two directories *that are both present*, at any given time, on the filesystem. Here I move things around, and what I want to check is that the content is preserved across moves (or that I moved the right thing in the right place).